middleware/src/hsm/hsm.go

```diff
@@ -203,3 +204,18 @@ func (hsm *HSM) InteractWithBootloader(f func(*bb02bootloader.Device)) error {
 	f(device)
 	return nil
 }
+
+// UpdateFirmware reboots into Bootloader and executes the firmware update
+func (hsm *HSM) UpdateFirmware(hsmFirmwareFile string) error {
```

Comment:
nit: `Upgrade` over `Update` as this word is also used in the bootloader and everywhere else.